local_net/in-place-fork.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+source ./utils.sh
+
+alias nobled=../build/nobled
+
+# Handle termination gracefully
+cleanup() {
+    echo "Stopping processes..."
+    kill "$NOBLED_PID" "$SYNC_PID" "$TAIL_PID" "$NOBLED_PID2" 2>/dev/null
+    exit 0
+}
+trap cleanup SIGINT SIGTERM
+
+HOME1=.duke
+
+CHAINID="noble-1"
+PEERS="4f9df51e0800e79e0d45fd376c11236b99be4e12@99.79.58.157:26656,3402e50ad4d838b26f8341a956c7b4b8a3c61ee5@65.109.93.44:21556"
+SNAP_RPC="https://noble-rpc.polkachu.com:443"
+GENESIS="https://raw.githubusercontent.com/noble-assets/networks/refs/heads/main/mainnet/noble-1/genesis.json"
+
+UPGRADE_ARG=""
+
+for arg in "$@"; do
+    case $arg in
+        -r|--reset)
+            rm -rf "$HOME1"
+            ;;
+        -t|--testnet)
+            CHAINID="grand-1"
+            PEERS="f2067cc7a23a4b2525f5f98430797b1e5c92e3aa@35.183.110.236:26656,8b22414f37d381a99ba99cd1edc5b884d43b7e53@65.109.23.114:21556"
+            SNAP_RPC="https://noble-testnet-rpc.polkachu.com:443"
+            GENESIS="https://raw.githubusercontent.com/noble-assets/networks/refs/heads/main/testnet/grand-1/genesis.json"
+            ;;
+        -u|--trigger-testnet-upgrade)
```

Comment:
I honestly think `--upgrade` is fine, as passing that in but running the script for mainnet might be confusing to whoever is using this!

```suggestion
        -u|--upgrade)
```

Reply:
I'd like to keep as is as this aligns with the actual flag that is being run by the noble binary. But feel free to commit your suggestion and override me.

Reply:
Okay! That's fair enough, leaving as is 😄